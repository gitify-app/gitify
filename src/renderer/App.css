html,
body,
#root {
  height: 100%;
  -webkit-user-select: none;
}

html::-webkit-scrollbar,
div.flex-grow::-webkit-scrollbar {
  width: 10px;
}

html::-webkit-scrollbar-thumb,
div.flex-grow::-webkit-scrollbar-thumb {
  background-color: #c1c1c1;
  border-radius: 10px;
}

html::-webkit-scrollbar-thumb:hover,
div.flex-grow::-webkit-scrollbar-thumb:hover {
  background-color: #a8a8a8;
  border-radius: 10px;
}

html.dark::-webkit-scrollbar-track,
html.dark div.flex-grow::-webkit-scrollbar-track {
  background-color: #090e15;
}

html.dark::-webkit-scrollbar-thumb,
html.dark div.flex-grow::-webkit-scrollbar-thumb {
  background-color: #24292e;
}

html.dark::-webkit-scrollbar-thumb:hover,
html.dark div.flex-grow::-webkit-scrollbar-thumb:hover {
  background-color: #3a3f44;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  margin: 0;
  cursor: default;
}

#nprogress {
  pointer-events: none;
}

#nprogress .bar {
  position: fixed;
  top: 0;
  left: 50px;
  z-index: 1031;
  background: #203354;

  width: 100%;
  height: 2px;
}

/**
 * Set emoji size according to surrounding text.
 * Ref: https://github.com/jdecked/twemoji?tab=readme-ov-file#inline-styles
 */
img.emoji {
  height: 1em;
  width: 1em;
  margin: 0 0.05em 0 0.1em;
  vertical-align: -0.1em;
}
