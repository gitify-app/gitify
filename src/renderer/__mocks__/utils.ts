import * as helpers from '../utils/helpers';

export function mockDirectoryPath() {
  jest.spyOn(helpers, 'getDirectoryPath').mockReturnValue('/mocked/dir/name');
}

/**
 * Ensure stable snapshots for our randomized emoji use-cases
 */
export function ensureStableEmojis() {
  global.Math.random = jest.fn(() => 0.1);
}
