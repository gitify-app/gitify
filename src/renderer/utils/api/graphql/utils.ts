export function formatAsGitHubSearchSyntax(
  repo: string,
  title: string,
): string {
  return `${title} in:title repo:${repo}`;
}
