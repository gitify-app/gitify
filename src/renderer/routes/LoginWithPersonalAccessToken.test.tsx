import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { AppContext } from '../context/App';
import * as comms from '../utils/comms';
import {
  LoginWithPersonalAccessTokenRoute,
  validateForm,
} from './LoginWithPersonalAccessToken';

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('renderer/routes/LoginWithPersonalAccessToken.tsx', () => {
  const mockLoginWithPersonalAccessToken = jest.fn();
  const openExternalLinkMock = jest
    .spyOn(comms, 'openExternalLink')
    .mockImplementation();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const tree = render(
      <MemoryRouter>
        <LoginWithPersonalAccessTokenRoute />
      </MemoryRouter>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('let us go back', () => {
    render(
      <MemoryRouter>
        <LoginWithPersonalAccessTokenRoute />
      </MemoryRouter>,
    );

    fireEvent.click(screen.getByTestId('header-nav-back'));

    expect(mockNavigate).toHaveBeenNthCalledWith(1, -1);
  });

  it('should validate the form values', () => {
    const emptyValues = {
      hostname: null,
      token: null,
    };

    let values = {
      ...emptyValues,
    };
    expect(validateForm(values).hostname).toBe('Hostname is required');
    expect(validateForm(values).token).toBe('Token is required');

    values = {
      ...emptyValues,
      hostname: 'hello',
      token: '!@Â£INVALID-.1',
    };
    expect(validateForm(values).hostname).toBe('Hostname format is invalid');
    expect(validateForm(values).token).toBe('Token format is invalid');
  });

  describe("'Generate a PAT' button", () => {
    it('should be disabled if no hostname configured', async () => {
      render(
        <AppContext.Provider
          value={{
            loginWithPersonalAccessToken: mockLoginWithPersonalAccessToken,
          }}
        >
          <MemoryRouter>
            <LoginWithPersonalAccessTokenRoute />
          </MemoryRouter>
        </AppContext.Provider>,
      );

      fireEvent.change(screen.getByTestId('login-hostname'), {
        target: { value: '' },
      });

      fireEvent.click(screen.getByTestId('login-create-token'));

      expect(openExternalLinkMock).toHaveBeenCalledTimes(0);
    });

    it('should open in browser if hostname configured', async () => {
      render(
        <AppContext.Provider
          value={{
            loginWithPersonalAccessToken: mockLoginWithPersonalAccessToken,
          }}
        >
          <MemoryRouter>
            <LoginWithPersonalAccessTokenRoute />
          </MemoryRouter>
        </AppContext.Provider>,
      );

      fireEvent.click(screen.getByTestId('login-create-token'));

      expect(openExternalLinkMock).toHaveBeenCalledTimes(1);
    });
  });

  it('should login using a token - success', async () => {
    mockLoginWithPersonalAccessToken.mockResolvedValueOnce(null);

    render(
      <AppContext.Provider
        value={{
          loginWithPersonalAccessToken: mockLoginWithPersonalAccessToken,
        }}
      >
        <MemoryRouter>
          <LoginWithPersonalAccessTokenRoute />
        </MemoryRouter>
      </AppContext.Provider>,
    );

    fireEvent.change(screen.getByTestId('login-hostname'), {
      target: { value: 'github.com' },
    });

    fireEvent.change(screen.getByTestId('login-token'), {
      target: { value: '1234567890123456789012345678901234567890' },
    });

    fireEvent.click(screen.getByTestId('login-submit'));

    await waitFor(() =>
      expect(mockLoginWithPersonalAccessToken).toHaveBeenCalledTimes(1),
    );

    expect(mockLoginWithPersonalAccessToken).toHaveBeenCalledTimes(1);
    expect(mockNavigate).toHaveBeenNthCalledWith(1, -1);
  });

  it('should login using a token - failure', async () => {
    mockLoginWithPersonalAccessToken.mockRejectedValueOnce(null);

    render(
      <AppContext.Provider
        value={{
          loginWithPersonalAccessToken: mockLoginWithPersonalAccessToken,
        }}
      >
        <MemoryRouter>
          <LoginWithPersonalAccessTokenRoute />
        </MemoryRouter>
      </AppContext.Provider>,
    );

    fireEvent.change(screen.getByTestId('login-hostname'), {
      target: { value: 'github.com' },
    });

    fireEvent.change(screen.getByTestId('login-token'), {
      target: { value: '1234567890123456789012345678901234567890' },
    });

    fireEvent.click(screen.getByTestId('login-submit'));

    await waitFor(() =>
      expect(mockLoginWithPersonalAccessToken).toHaveBeenCalledTimes(1),
    );

    expect(mockLoginWithPersonalAccessToken).toHaveBeenCalledTimes(1);
    expect(mockNavigate).toHaveBeenCalledTimes(0);
  });

  it('should render the form with errors', () => {
    render(
      <MemoryRouter>
        <LoginWithPersonalAccessTokenRoute />
      </MemoryRouter>,
    );

    fireEvent.change(screen.getByTestId('login-hostname'), {
      target: { value: 'test' },
    });
    fireEvent.change(screen.getByTestId('login-token'), {
      target: { value: '123' },
    });

    fireEvent.click(screen.getByTestId('login-submit'));

    expect(screen.getByTestId('login-errors')).toBeTruthy();
    expect(screen.getByText('Hostname format is invalid')).toBeTruthy();
    expect(screen.getByText('Token format is invalid')).toBeTruthy();
  });

  it('should open help docs in the browser', async () => {
    render(
      <AppContext.Provider
        value={{
          loginWithPersonalAccessToken: mockLoginWithPersonalAccessToken,
        }}
      >
        <MemoryRouter>
          <LoginWithPersonalAccessTokenRoute />
        </MemoryRouter>
      </AppContext.Provider>,
    );

    fireEvent.click(screen.getByTestId('login-docs'));

    expect(openExternalLinkMock).toHaveBeenCalledTimes(1);
  });
});
