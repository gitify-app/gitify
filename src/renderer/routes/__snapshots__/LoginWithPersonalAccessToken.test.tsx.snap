// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer/routes/LoginWithPersonalAccessToken.tsx renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-6 mt-2 py-2"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 dStkQv"
          data-align="stretch"
          data-direction="horizontal"
          data-justify="space-between"
          data-padding="none"
          data-wrap="nowrap"
        >
          <button
            aria-describedby=":r0:-loading-announcement"
            aria-labelledby="Go Back"
            class="types__StyledButton-sc-ws60qy-0 ggqBNv"
            data-component="IconButton"
            data-loading="false"
            data-no-visuals="true"
            data-testid="header-nav-back"
            sx="[object Object]"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-arrow-left"
              display="inline-block"
              fill="currentColor"
              focusable="false"
              height="16"
              overflow="visible"
              style="vertical-align: text-bottom;"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z"
              />
            </svg>
          </button>
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 dStkQv"
            data-align="center"
            data-direction="horizontal"
            data-gap="condensed"
            data-justify="start"
            data-padding="none"
            data-wrap="nowrap"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-key Octicon-sc-9kayk9-0"
              display="inline-block"
              fill="currentColor"
              focusable="false"
              height="18"
              overflow="visible"
              style="vertical-align: text-bottom;"
              viewBox="0 0 16 16"
              width="18"
            >
              <path
                d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"
              />
            </svg>
            <h2
              class="Heading__StyledHeading-sc-1c1dgg0-0 iLDWTC"
              sx="[object Object]"
            >
              Login with Personal Access Token
            </h2>
          </div>
        </div>
      </div>
      <div
        class="px-8"
      >
        <form>
          <div
            class="mb-4"
          >
            <label
              class="mb-2 block text-sm font-semibold tracking-wide"
              for="hostname"
            >
              Hostname
            </label>
            <input
              class="mb-2 block w-full appearance-none rounded border bg-gray-100 px-4 py-1.5 text-sm focus:bg-gray-200 focus:outline-none dark:text-gray-800 border-gray-500"
              id="hostname"
              name="hostname"
              placeholder="github.company.com"
              type="text"
              value="github.com"
            />
            <div
              class="mt-3 text-xs text-gitify-font"
            >
              <div
                class="Stack__StyledStack-sc-x3xa2i-0 dStkQv"
                data-align="stretch"
                data-direction="vertical"
                data-gap="condensed"
                data-justify="start"
                data-padding="none"
                data-wrap="nowrap"
              >
                <i
                  class="Text-sc-17v1xeu-0 gCQeDW"
                >
                  Change only if you are using GitHub Enterprise Server.
                </i>
                <div
                  class="Stack__StyledStack-sc-x3xa2i-0 dStkQv"
                  data-align="center"
                  data-direction="horizontal"
                  data-gap="condensed"
                  data-justify="start"
                  data-padding="none"
                  data-wrap="nowrap"
                >
                  <button
                    aria-describedby=":r1:-loading-announcement"
                    class="types__StyledButton-sc-ws60qy-0 bBRiTk"
                    data-loading="false"
                    data-size="small"
                    data-testid="login-create-token"
                    sx="[object Object]"
                    type="button"
                  >
                    <span
                      class="Box-sc-g0xbh4-0 lbSBsX"
                      data-component="buttonContent"
                      sx="[object Object]"
                    >
                      <span
                        class="Box-sc-g0xbh4-0 bPvALJ"
                        data-component="leadingVisual"
                        sx="[object Object]"
                      >
                        <svg
                          aria-hidden="true"
                          class="octicon octicon-key"
                          display="inline-block"
                          fill="currentColor"
                          focusable="false"
                          height="16"
                          overflow="visible"
                          style="vertical-align: text-bottom;"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <path
                            d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"
                          />
                        </svg>
                      </span>
                      <span
                        data-component="text"
                      >
                        Generate a PAT
                      </span>
                    </span>
                  </button>
                  <span
                    class="Text-sc-17v1xeu-0 gCQeDW"
                  >
                    on GitHub then paste your token below.
                  </span>
                </div>
                <i
                  class="Text-sc-17v1xeu-0 gCQeDW"
                >
                  The required scopes will be automatically selected for you.
                </i>
              </div>
            </div>
          </div>
          <div
            class="mb-4"
          >
            <label
              class="mb-2 block text-sm font-semibold tracking-wide"
              for="token"
            >
              Token
            </label>
            <input
              class="mb-2 block w-full appearance-none rounded border bg-gray-100 px-4 py-1.5 text-sm focus:bg-gray-200 focus:outline-none dark:text-gray-800 border-red-500"
              id="token"
              name="token"
              placeholder="The 40 characters token generated on GitHub"
              type="text"
              value=""
            />
          </div>
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 dStkQv"
            data-align="center"
            data-direction="horizontal"
            data-justify="space-between"
            data-padding="none"
            data-wrap="nowrap"
          >
            <span
              aria-label="GitHub documentation"
              class="Tooltip__TooltipBase-sc-17tf59c-0 rtmEP tooltipped-n"
              id=":r2:"
              role="tooltip"
            >
              <button
                aria-describedby=":r3:-loading-announcement"
                class="types__StyledButton-sc-ws60qy-0 bBRiTk"
                data-loading="false"
                data-size="small"
                data-testid="login-docs"
                sx="[object Object]"
                type="button"
              >
                <span
                  class="Box-sc-g0xbh4-0 lbSBsX"
                  data-component="buttonContent"
                  sx="[object Object]"
                >
                  <span
                    class="Box-sc-g0xbh4-0 bPvALJ"
                    data-component="leadingVisual"
                    sx="[object Object]"
                  >
                    <svg
                      aria-hidden="true"
                      class="octicon octicon-book"
                      display="inline-block"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      overflow="visible"
                      style="vertical-align: text-bottom;"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <path
                        d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z"
                      />
                    </svg>
                  </span>
                  <span
                    data-component="text"
                  >
                    Docs
                  </span>
                </span>
              </button>
            </span>
            <span
              aria-label="Login"
              class="Tooltip__TooltipBase-sc-17tf59c-0 rtmEP tooltipped-n"
              id=":r4:"
              role="tooltip"
            >
              <button
                aria-describedby=":r5:-loading-announcement"
                class="types__StyledButton-sc-ws60qy-0 bBXWFS"
                data-loading="false"
                data-testid="login-submit"
                disabled=""
                sx="[object Object]"
                type="submit"
              >
                <span
                  class="Box-sc-g0xbh4-0 lbSBsX"
                  data-component="buttonContent"
                  sx="[object Object]"
                >
                  <span
                    class="Box-sc-g0xbh4-0 bPvALJ"
                    data-component="leadingVisual"
                    sx="[object Object]"
                  >
                    <svg
                      aria-hidden="true"
                      class="octicon octicon-sign-in"
                      display="inline-block"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      overflow="visible"
                      style="vertical-align: text-bottom;"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <path
                        d="M2 2.75C2 1.784 2.784 1 3.75 1h2.5a.75.75 0 0 1 0 1.5h-2.5a.25.25 0 0 0-.25.25v10.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 0 1.5h-2.5A1.75 1.75 0 0 1 2 13.25Zm6.56 4.5h5.69a.75.75 0 0 1 0 1.5H8.56l1.97 1.97a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L6.22 8.53a.75.75 0 0 1 0-1.06l3.25-3.25a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734Z"
                      />
                    </svg>
                  </span>
                  <span
                    data-component="text"
                  >
                    Login
                  </span>
                </span>
              </button>
            </span>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-6 mt-2 py-2"
    >
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 dStkQv"
        data-align="stretch"
        data-direction="horizontal"
        data-justify="space-between"
        data-padding="none"
        data-wrap="nowrap"
      >
        <button
          aria-describedby=":r0:-loading-announcement"
          aria-labelledby="Go Back"
          class="types__StyledButton-sc-ws60qy-0 ggqBNv"
          data-component="IconButton"
          data-loading="false"
          data-no-visuals="true"
          data-testid="header-nav-back"
          sx="[object Object]"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-arrow-left"
            display="inline-block"
            fill="currentColor"
            focusable="false"
            height="16"
            overflow="visible"
            style="vertical-align: text-bottom;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z"
            />
          </svg>
        </button>
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 dStkQv"
          data-align="center"
          data-direction="horizontal"
          data-gap="condensed"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-key Octicon-sc-9kayk9-0"
            display="inline-block"
            fill="currentColor"
            focusable="false"
            height="18"
            overflow="visible"
            style="vertical-align: text-bottom;"
            viewBox="0 0 16 16"
            width="18"
          >
            <path
              d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"
            />
          </svg>
          <h2
            class="Heading__StyledHeading-sc-1c1dgg0-0 iLDWTC"
            sx="[object Object]"
          >
            Login with Personal Access Token
          </h2>
        </div>
      </div>
    </div>
    <div
      class="px-8"
    >
      <form>
        <div
          class="mb-4"
        >
          <label
            class="mb-2 block text-sm font-semibold tracking-wide"
            for="hostname"
          >
            Hostname
          </label>
          <input
            class="mb-2 block w-full appearance-none rounded border bg-gray-100 px-4 py-1.5 text-sm focus:bg-gray-200 focus:outline-none dark:text-gray-800 border-gray-500"
            id="hostname"
            name="hostname"
            placeholder="github.company.com"
            type="text"
            value="github.com"
          />
          <div
            class="mt-3 text-xs text-gitify-font"
          >
            <div
              class="Stack__StyledStack-sc-x3xa2i-0 dStkQv"
              data-align="stretch"
              data-direction="vertical"
              data-gap="condensed"
              data-justify="start"
              data-padding="none"
              data-wrap="nowrap"
            >
              <i
                class="Text-sc-17v1xeu-0 gCQeDW"
              >
                Change only if you are using GitHub Enterprise Server.
              </i>
              <div
                class="Stack__StyledStack-sc-x3xa2i-0 dStkQv"
                data-align="center"
                data-direction="horizontal"
                data-gap="condensed"
                data-justify="start"
                data-padding="none"
                data-wrap="nowrap"
              >
                <button
                  aria-describedby=":r1:-loading-announcement"
                  class="types__StyledButton-sc-ws60qy-0 bBRiTk"
                  data-loading="false"
                  data-size="small"
                  data-testid="login-create-token"
                  sx="[object Object]"
                  type="button"
                >
                  <span
                    class="Box-sc-g0xbh4-0 lbSBsX"
                    data-component="buttonContent"
                    sx="[object Object]"
                  >
                    <span
                      class="Box-sc-g0xbh4-0 bPvALJ"
                      data-component="leadingVisual"
                      sx="[object Object]"
                    >
                      <svg
                        aria-hidden="true"
                        class="octicon octicon-key"
                        display="inline-block"
                        fill="currentColor"
                        focusable="false"
                        height="16"
                        overflow="visible"
                        style="vertical-align: text-bottom;"
                        viewBox="0 0 16 16"
                        width="16"
                      >
                        <path
                          d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"
                        />
                      </svg>
                    </span>
                    <span
                      data-component="text"
                    >
                      Generate a PAT
                    </span>
                  </span>
                </button>
                <span
                  class="Text-sc-17v1xeu-0 gCQeDW"
                >
                  on GitHub then paste your token below.
                </span>
              </div>
              <i
                class="Text-sc-17v1xeu-0 gCQeDW"
              >
                The required scopes will be automatically selected for you.
              </i>
            </div>
          </div>
        </div>
        <div
          class="mb-4"
        >
          <label
            class="mb-2 block text-sm font-semibold tracking-wide"
            for="token"
          >
            Token
          </label>
          <input
            class="mb-2 block w-full appearance-none rounded border bg-gray-100 px-4 py-1.5 text-sm focus:bg-gray-200 focus:outline-none dark:text-gray-800 border-red-500"
            id="token"
            name="token"
            placeholder="The 40 characters token generated on GitHub"
            type="text"
            value=""
          />
        </div>
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 dStkQv"
          data-align="center"
          data-direction="horizontal"
          data-justify="space-between"
          data-padding="none"
          data-wrap="nowrap"
        >
          <span
            aria-label="GitHub documentation"
            class="Tooltip__TooltipBase-sc-17tf59c-0 rtmEP tooltipped-n"
            id=":r2:"
            role="tooltip"
          >
            <button
              aria-describedby=":r3:-loading-announcement"
              class="types__StyledButton-sc-ws60qy-0 bBRiTk"
              data-loading="false"
              data-size="small"
              data-testid="login-docs"
              sx="[object Object]"
              type="button"
            >
              <span
                class="Box-sc-g0xbh4-0 lbSBsX"
                data-component="buttonContent"
                sx="[object Object]"
              >
                <span
                  class="Box-sc-g0xbh4-0 bPvALJ"
                  data-component="leadingVisual"
                  sx="[object Object]"
                >
                  <svg
                    aria-hidden="true"
                    class="octicon octicon-book"
                    display="inline-block"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    overflow="visible"
                    style="vertical-align: text-bottom;"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <path
                      d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z"
                    />
                  </svg>
                </span>
                <span
                  data-component="text"
                >
                  Docs
                </span>
              </span>
            </button>
          </span>
          <span
            aria-label="Login"
            class="Tooltip__TooltipBase-sc-17tf59c-0 rtmEP tooltipped-n"
            id=":r4:"
            role="tooltip"
          >
            <button
              aria-describedby=":r5:-loading-announcement"
              class="types__StyledButton-sc-ws60qy-0 bBXWFS"
              data-loading="false"
              data-testid="login-submit"
              disabled=""
              sx="[object Object]"
              type="submit"
            >
              <span
                class="Box-sc-g0xbh4-0 lbSBsX"
                data-component="buttonContent"
                sx="[object Object]"
              >
                <span
                  class="Box-sc-g0xbh4-0 bPvALJ"
                  data-component="leadingVisual"
                  sx="[object Object]"
                >
                  <svg
                    aria-hidden="true"
                    class="octicon octicon-sign-in"
                    display="inline-block"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    overflow="visible"
                    style="vertical-align: text-bottom;"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <path
                      d="M2 2.75C2 1.784 2.784 1 3.75 1h2.5a.75.75 0 0 1 0 1.5h-2.5a.25.25 0 0 0-.25.25v10.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 0 1.5h-2.5A1.75 1.75 0 0 1 2 13.25Zm6.56 4.5h5.69a.75.75 0 0 1 0 1.5H8.56l1.97 1.97a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L6.22 8.53a.75.75 0 0 1 0-1.06l3.25-3.25a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734Z"
                    />
                  </svg>
                </span>
                <span
                  data-component="text"
                >
                  Login
                </span>
              </span>
            </button>
          </span>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
